backend/onedrive/onedrive.go

```diff
@@ -1503,7 +1503,50 @@ func (f *Fs) PublicLink(ctx context.Context, remote string, expire fs.Duration,
 		fmt.Println(err)
 		return "", err
 	}
-	return result.Link.WebURL, nil
+
+	url := result.Link.WebURL
+
+	// Convert share link to direct download link if target is not a folder
+	segments := strings.Split(url, "/")
+	cnvFailMsg := "Conversion method is outdated. Cannot convert to downloadable URL"
+	switch {
+	case f.driveType == driveTypePersonal:
+		// https://stackoverflow.com/questions/37951114/direct-download-link-to-onedrive-file
+		if len(segments) != 5 {
+			fs.Logf(f, cnvFailMsg)
+		} else if segments[3] != "u" {
+			enc := base64.StdEncoding.EncodeToString([]byte(url))
+			url = "https://api.onedrive.com/v1.0/shares/u!" + enc[:len(enc)-1] + "/root/content"
+		}
+	case f.driveType == driveTypeBusiness:
+		// https://docs.microsoft.com/en-us/sharepoint/dev/spfx/shorter-share-link-format
+		if len(segments) != 8 {
+			fs.Logf(f, cnvFailMsg)
```

Comment:
Generally I think [early returning](https://danp.net/posts/reducing-go-nesting/) is preferred when there's something failing and we don't have anything else to do afterwards. It helps with the readability. (The same applies to others)

Reply:
Sounds like a good idea.